monai/fl/client/monai_algo.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import sys
+from typing import TYPE_CHECKING, Optional
+
+import torch
+
+import monai
+from monai.bundle import ConfigParser
+from monai.config import IgniteInfo
+from monai.fl.client.client_algo import ClientAlgo
+from monai.fl.utils.constants import ExtraItems, FlPhase, WeightType
+from monai.fl.utils.exchange_object import ExchangeObject
+from monai.utils import min_version, optional_import
+
+if TYPE_CHECKING:
+    from ignite.engine import Events
+else:
+    Events, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Events")
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO, format="%(asctime)s - %(message)s")
+
+
+def convert_global_weights(global_weights, local_var_dict):
+    """Helper function to convert global weights to local weights format"""
+    # Before loading weights, tensors might need to be reshaped to support HE for secure aggregation.
+    model_keys = global_weights.keys()
+    for var_name in local_var_dict:
+        if var_name in model_keys:
+            try:
+                # update the local dict
+                local_var_dict[var_name] = torch.as_tensor(global_weights[var_name])
+            except Exception as e:
+                raise ValueError(f"Convert weight from {var_name} failed.") from e
+    return local_var_dict
+
+
+class MonaiAlgo(ClientAlgo):
```

Comment:
Could you please help add the doc-string for this class and the functions?

Thanks in advance.

Reply:
Will be addressed in next PR to solve issue #4845